packages/preact-iso/router.js

```diff
@@ -1,16 +1,34 @@
 import { h, createContext, cloneElement } from 'preact';
 import { useContext, useMemo, useReducer, useEffect, useLayoutEffect, useRef } from 'preact/hooks';
 
-const UPDATE = (state, url, push) => {
-	if (url && url.type === 'click') {
+/**
+ * @param {string} state
+ * @param {MouseEvent|PopStateEvent|Object|string} url
+ * @return {string}
+ */
+const UPDATE = (state, url) => {
+	/** @type {boolean|undefined} - History state update strategy */
+	let push = undefined
+
+	// user click
+	if (url instanceof MouseEvent) {
```

Comment:
This is too restrictive. It will fail for cross-document events or synthetic events. I know the duck-typing that was here wasn't great from a TS perspective, but it's the best option we have for compatibility and size and we should keep it.

Reply:
okay